### DISCUSSION TITLE
1 pgs not scrubbed of the pool "device_health_metrics"?

### DISCUSSION THREAD

**Discussion by zhucan** [rook/rook#10733] [Q&A]
Reproduce: 

1. create  a new  crush root
2. create a new crush host
3. move the osd from old crush host to new crush host
4. add the new crush host  to new crush root

Beause of we don't want to use the default crush root to crate pool. 

![image](https://user-images.githubusercontent.com/13997135/185284656-76cc957d-4f5c-4942-b599-9072936d8aa4.png)

I know how to fix it, only create a new crush rule from new crush root and config the pool "device_health_metrics", but we don't want to fix it manually, is there a way to fix it?

**Comment by travisn** [comment]
You can configure the built-in pool "device_health_metrics" with the pool CR as seen in [this example](https://github.com/rook/rook/blob/release-1.8/deploy/examples/cluster-test.yaml#L55-L65) (or for Quincy the pool was renamed to `.mgr` as seen in [this example](https://github.com/rook/rook/blob/release-1.9/deploy/examples/cluster-test.yaml#L58-L67)). Have you tried that?

**Reply by zhucan** [reply]
The replication is 1? The ceph status show "1 pool(s) have no replicas configured".

**Reply by zhucan** [reply]
Beacuse of the pool 'device_health_metrics', we can not create a new crush root use exists osd(the pgs on it).

**Reply by travisn** [reply]
In that example I linked above, you can change the replication to 3, or any other settings such as the `crushRoot`.
If Ceph creates the pool first, you can try deleting the pool, then create that pool CR with the new crushRoot.

**Reply by zhucan** [reply]
yes. we need to delete the pool firstly and then create the same pool with new crushRoot. Is it possible to disable or enable the ceph to create the pool by user in a later version.

**Reply by travisn** [reply]
I don't believe there is a way to tell Ceph not to automatically create that pool, so you may just need to delete it so it can be reconfigured.

**Reply by zhucan** [reply]
it's possible to add a config to ceph that we can enable or disable it.

**Comment by zhucan** [comment]
I will try it

**Comment by zhucan** [comment]
@travisn If we add the disk to the ceph cluster cr one by one after the ceph cluster was created
ceph osd tree:
[root@rook-node02 /]# ceph osd tree
ID  CLASS  WEIGHT  TYPE NAME     STATUS  REWEIGHT  PRI-AFF
-1              0  root default                           
 0    hdd       0  osd.0             up   1.00000  1.00000
 1    hdd       0  osd.1             up   1.00000  1.00000
 2    hdd       0  osd.2             up   1.00000  1.00000
 3    hdd       0  osd.3             up   1.00000  1.00000
 4    hdd       0  osd.4             up   1.00000  1.00000
 5    hdd       0  osd.5             up   1.00000  1.00000

ceph cluster status is:
[root@rook-node02 /]# ceph -s 
  cluster:
    id:     90b22732-453a-4fbb-b146-d9f0334fc2b3
    health: HEALTH_WARN
            mon a is low on available space
            Reduced data availability: 32 pgs inactive
            1 pool(s) have no replicas configured
 
  services:
    mon: 3 daemons, quorum a,b,c (age 17m)
    mgr: a(active, since 17m), standbys: b
    osd: 6 osds: 6 up (since 53s), 6 in (since 65s)
 
  data:
    pools:   1 pools, 32 pgs
    objects: 0 objects, 0 B
    usage:   30 MiB used, 600 GiB / 600 GiB avail
    pgs:     100.000% pgs unknown
             32 unknown

**Comment by zhucan** [comment]
if we set "osd crush update on start = false" to the configmap of “rook-config-override”， the  osd tree print:
[root@rook-node02 /]# ceph osd tree
ID CLASS WEIGHT TYPE NAME STATUS REWEIGHT PRI-AFF
-1 0 root default
0 hdd 0 osd.0 up 1.00000 1.00000
1 hdd 0 osd.1 up 1.00000 1.00000
2 hdd 0 osd.2 up 1.00000 1.00000
3 hdd 0 osd.3 up 1.00000 1.00000
4 hdd 0 osd.4 up 1.00000 1.00000
5 hdd 0 osd.5 up 1.00000 1.00000

we only want to keep the osd tree the same with pre after the node reboot. Is there a way to deal with it?
